tensorflow/java/src/gen/perl/tftypes.pl

```diff
@@ -76,42 +91,51 @@ sub usage {
         # Generate class declarations
         # print STDERR "Creating $dirname/$tfname.java\n";
         open (CLASSFILE, ">$dirname/$tfname.java") || die "Can't open $tfname.java";
-        print CLASSFILE "// GENERATED FILE. Edit tftypes.pl instead.\n";
-        print CLASSFILE "package org.tensorflow.types;\n\n";
-        print CLASSFILE  "/** The class $tfname represents $desc. */\n"
-                        ."public class $tfname implements Types.TFType {\n"
-                        ."  /** Represents the type $tfname at run time. */\n"
-                        ."  public static final Class<$tfname> T = $tfname.class;\n"
+        print CLASSFILE
+'/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+// GENERATED FILE. To update, edit tftypes.pl instead.
+
+';
+        my $fulldesc = $desc;
+        if (substr($desc, 0, 1) =~ m/^[aeoiu8]$/i) {
+            $fulldesc = "an $desc"
+        } else {
+            $fulldesc = "a $desc"
+        }
+        print CLASSFILE  "package org.tensorflow.types;\n\n";
+        print CLASSFILE  "/** The class $tfname represents $fulldesc. */\n"
+                        ."public class $tfname implements TFType {\n"
                         ."  static {\n"
-                        ."    Types.typeCodes.put($tfname.T, $index);\n"
+                        ."    Types.typeCodes.put($tfname.class, org.tensorflow.DataType.$ucname);\n"
```

Comment:
Out of curiosity: Why the fully qualified name here instead of an import + `DataType.$ucname` here?
(I don't have a strong opinion either way, I just thought the import would be more idiomatic, but either is fine)